common/execution_context.go

```diff
@@ -273,33 +274,48 @@ func (e *ExecutionContext) eval(
 	return res, nil
 }
 
+// Based on: https://github.com/microsoft/playwright/blob/master/src/server/injected/injectedScript.ts
+//go:embed js/injected_script.js
+var injectedScriptSource string
+
 // getInjectedScript returns a JS handle to the injected script of helper functions.
 func (e *ExecutionContext) getInjectedScript(apiCtx context.Context) (api.JSHandle, error) {
 	e.logger.Debugf(
 		"ExecutionContext:getInjectedScript",
 		"sid:%s stid:%s fid:%s ectxid:%d efurl:%s",
 		e.sid, e.stid, e.fid, e.id, e.furl)
 
-	if e.injectedScript == nil {
-		rt := k6common.GetRuntime(e.ctx)
-		suffix := `//# sourceURL=` + evaluationScriptURL
-		source := fmt.Sprintf(`(() => {%s; return new InjectedScript();})()`, injectedScriptSource)
-		expression := source
-		expressionWithSourceURL := expression
-		if !sourceURLRegex.Match([]byte(expression)) {
-			expressionWithSourceURL = expression + "\n" + suffix
-		}
+	if e.injectedScript != nil {
+		return e.injectedScript, nil
+	}
 
-		opts := evalOptions{
-			forceCallable: false,
-			returnByValue: false,
-		}
-		handle, err := e.eval(apiCtx, opts, rt.ToValue(expressionWithSourceURL))
-		if handle == nil || err != nil {
-			return nil, fmt.Errorf("cannot get injected script (%q): %w", suffix, err)
-		}
-		e.injectedScript = handle.(api.JSHandle)
+	var (
+		rt                      = k6common.GetRuntime(e.ctx)
+		suffix                  = `//# sourceURL=` + evaluationScriptURL
+		source                  = fmt.Sprintf(`(() => {%s; return new InjectedScript();})()`, injectedScriptSource)
+		expression              = source
+		expressionWithSourceURL = expression
+	)
+	if !sourceURLRegex.Match([]byte(expression)) {
+		expressionWithSourceURL = expression + "\n" + suffix
+	}
+	handle, err := e.eval(
+		apiCtx,
+		evalOptions{forceCallable: false, returnByValue: false},
+		rt.ToValue(expressionWithSourceURL),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("cannot get injected script (%q): %w", suffix, err)
+	}
+	if handle == nil {
+		return nil, fmt.Errorf("cannot get handle for inject script (%q): %w", suffix, err)
```

Comment:
`err` will be nil here, so:
```suggestion
		return nil, fmt.Errorf("cannot get handle for injected script (%q)", suffix)
```

Though I'm wondering why we include the `suffix`. Isn't it always `//# sourceURL=__xk6_browser_evaluation_script__`?

Reply:
Yeah, it's not helpful to include it there. It's constant, after all.

Reply:
Il est normal que vous ayez déjà été vu car cela fait partie d'un grand PR. I mean, don't worry, that's expected because this PR is part of a larger PR :)

Reply:
Should be resolved in [6717f05](https://github.com/grafana/xk6-browser/pull/221/commits/6717f0592fc2b63175273aee922a13d8d450e06b).